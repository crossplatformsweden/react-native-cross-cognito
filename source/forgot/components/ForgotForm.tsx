import React from 'react';
import { View } from 'react-native';

export class ForgotForm extends React.Component {
    render() {
        return <View />;
    }
}

export default ForgotForm;
