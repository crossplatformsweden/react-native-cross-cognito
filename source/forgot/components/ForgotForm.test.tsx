// https://itnext.io/testing-react-16-3-components-with-react-test-renderer-without-enzyme-d9c65d689e88
import React from 'react';
import TestRenderer from 'react-test-renderer';
import ForgotForm from './ForgotForm';

jest.unmock('react-native');
jest.unmock('./ForgotForm');

// Default export requires this type of mocking
// jest.mock('react-native-stager', () => ({
//   __esModule: true,
//   Stage: 'View',
//   StageButtons: 'View',
//   StageProgress: 'View',
//   default: 'View',
// }));

describe('components', () => {
  describe('<ForgotForm />', () => {
    it('Component should render', () => {
      const wrapper = TestRenderer.create(
        <ForgotForm onSubmit={jest.fn()} onEmailChanged={jest.fn()} />
      );
      expect(wrapper.toJSON()).toMatchSnapshot();
    });

    // it('Should have paper <Button />', () => {
    //   const wrapper = TestRenderer.create(<ConfirmForm />);
    //   const child = wrapper.root.findByProps({ mode: 'text' });
    //   expect(child.type).toEqual('View');
    // });

    // it('<FontAwesomeButton /> onPress should be called', () => {
    //   let called = false;
    //   const onCalled = () => (called = !called);

    //   const wrapper = TestRenderer.create(<ConfirmForm />);
    //   const child = wrapper.root.findByProps({ name: 'map' });
    //   child.props.onPress();
    //   expect(called).toBeTruthy();
    // });
  });
});
