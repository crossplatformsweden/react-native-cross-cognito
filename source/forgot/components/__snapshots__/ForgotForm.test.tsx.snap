// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components <ForgotForm /> Component should render 1`] = `
<View
  style={
    Array [
      Object {},
    ]
  }
>
  <TextInput
    autoCapitalize="none"
    autoCorrect={false}
    clearButtonMode="always"
    label="E-mail"
    onChangeText={[Function]}
    placeholder="E-mail"
    value=""
  />
  <Button
    backgroundColor="#349b34"
    buttonStyle={
      Object {
        "margin": 0,
        "minHeight": 45,
      }
    }
    mode="contained"
    onPress={[MockFunction]}
    style={
      Object {
        "marginTop": 10,
      }
    }
    title="Reset password"
  />
</View>
`;
